routers/web/repo/setting_protected_branch.go

```diff
@@ -237,7 +238,22 @@ func SettingsProtectedBranchPost(ctx *context.Context) {
 
 	protectBranch.EnableStatusCheck = f.EnableStatusCheck
 	if f.EnableStatusCheck {
-		protectBranch.StatusCheckContexts = f.StatusCheckContexts
+		replacer := strings.NewReplacer("\r\n", "\n", "\r", "\n")
+		patterns := strings.Split(replacer.Replace(f.StatusCheckContexts), "\n")
+		validPatterns := make([]string, 0, len(patterns))
+		for _, c := range patterns {
+			trimmed := strings.TrimSpace(c)
+			if trimmed == "" {
+				continue
+			}
+			if _, err := glob.Compile(trimmed); err != nil {
+				ctx.Flash.Error(ctx.Tr("repo.settings.protect_invalid_status_check_pattern", c))
+				ctx.Redirect(fmt.Sprintf("%s/settings/branches/edit?rule_name=%s", ctx.Repo.RepoLink, protectBranch.RuleName))
```

Comment:
escape parameter

Reply:
fixed in 1645f25d5798a7123192c8d58a506e638c2ed7bc